src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -253,6 +250,13 @@ private async Task RecomputeTagsAsync(bool highPriority, CancellationToken cance
 
                     this.CachedTagTrees = newTagTrees;
                     this.State = context.State;
+                    if (this._subjectBuffer.CurrentSnapshot.Version.VersionNumber == subjectBufferVersion)
+                    {
+                        // Only clear the accumulated text changes if the subject buffer didn't change during the
+                        // tagging operation. Otherwise, it is impossible to know which changes occurred prior to the
+                        // request to tag, and which ones occurred during the tagging itself.
+                        this.AccumulatedTextChanges = null;
+                    }
```

Comment:
i'm not getting the logic here.

Reply:
This change is a separate commit. It might help to review it that way.

The underlying problem is we claimed data from the type _and reset that data_ at the point of acquisition. If cancellation occurs after acquisition but before processing completes, the accumulated text changes for that batch processing operation were simply discarded previously. The new code will re-process the accumulated text changes if cancellation interrupts processing.

Reply:
➡️ I retained this change as it is a correctness fix